### PR TITLE
Started adding IOS support

### PR DESCRIPTION
**Pull request by GGabriele**
Hi there.
This started as an 'exercise' for myself but I decided to make the PR since I'd like to contribute to this awesome project.

Working with IOS is a pain since it doesn't have native API, but I tried to do my best to meet the project requirements. It's far to be completed yet, but if you think this could be useful I'll keep working on it (like adding compare function and fine tuning the whole thing).

Let me know :)

### CODE REVIEW

**Comment by dbarrosop** [comment]
Hello @GGabriele,
Great job! Of course I think it would be useful : ) The only comment I will make about the code is don't do things like:

     def compare_config(self):
         pass

As you are not supporting config_replace I would suggest for starters to return whatever the config_merge is doing. Improving on that and returning the changes would be nice but given the limitations of the platform I am happy with compare_config returning just that. Either that or don't implement the method at all. That will trigger a NotImplementedException allowing the user of napalm to modify the workflow of his/her application depending on if a method for a particular platform is implemented or not.

Regards
David

**Comment by GGabriele** [comment]
Ehy @dbarrosop,
I made some changes as you suggested. Now the workflow is this:

- load_merge_candidate loads a candidate configuration (without sending/applying any commands)

- compare_config shows what changes would be applied from load_merge

- commit_config sends commands and save running configuration

- doing so, discard_config would simply look like a list of commands not yet applied and if it is called it empties the list. Anyway, if any error occours during the commit (like a typo into the configuration) the discard_config function is called and it will remove the commands that have been already applied with the 'no' prefix before the actual command

I've also added the get_lldp_neighbor function. Anyway, I did all my tests in GNS3 and since I didn't manage to find an IOS image supporting LLDP I used a textfile with a sample 'show lldp neighbors' output for my test.

File: napalm/ios.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+            device_type='cisco_ios',
+            ip=self.hostname,
+            username=self.username,
+            password=self.password)
+
+    def close(self):
+        """Closes the connection to the device."""
+        self.device.disconnect() 
+
+    def discard_config(self):
```
Comment (dbarrosop) [inline napalm/ios.py]: Quick question about this method. Are you trying to revert changes after they have been committed? If yes, you should do that with the rollback method. Discard should just discard the candidate configuration before commit. For example:

    device.open()
    device.load_merge_configuration(filename='my_changes.txt')
    # I changed my mind and I don't want to commit:
    device.discard_config()
    
    # Let's try this other instead
    device.load_merge_configuration(filename='my_other_changes.txt')
    device.commit()
    
    # Oh shit! I broke it : )
    device.rollback()
    device.close()

**Comment by dbarrosop** [comment]
Could you work on some documentation as well? Something that explains the specifics of the driver you wrote (considerations, caveats, anything you would like to know if you weren't the author and wanted to use the library). You can check others drivers' caveats.

**Comment by GGabriele** [comment]
Okay @dbarrosop, it definitely makes sense. 
I'm fixing it and adding docs :)

**Comment by dbarrosop** [comment]
Thanks! : )

**Comment by GGabriele** [comment]
I made the needed changes to the code and added some docs.
Let me know if everything is fine and if you need anything else :)

**Comment by dbarrosop** [comment]
Awesome work : )

**Comment by fooelisa** [comment]
Very nice!! :) +1

**Comment by GGabriele** [comment]
Thank you :) happy to contribute

**Pull request merged**